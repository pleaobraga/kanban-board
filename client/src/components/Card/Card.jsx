import './Card.scss'

import React from 'react'
import PropTypes from 'prop-types'
import { startCase, kebabCase } from 'lodash'
import { Draggable } from 'react-beautiful-dnd'
import { useDispatch } from 'react-redux'

import { deleteCard } from '../../reducers/BoardReducer'

export const Card = ({
  id,
  index,
  type,
  duration,
  severity,
  TaskListIndex
}) => {
  const dispatch = useDispatch()

  const deleteItem = () => {
    //TODO Implemen delete function

    const card = {
      id,
      index,
      TaskListIndex
    }

    dispatch(deleteCard(card))
  }

  return (
    <Draggable draggableId={id} index={index}>
      {(provided, snapshot) => (
        <div
          className={`card card--${kebabCase(type)} ${snapshot.isDragging &&
            'card--dragging'}`}
          ref={provided.innerRef}
          {...provided.draggableProps}
          {...provided.dragHandleProps}
        >
          <i className="material-icons close-icon" onClick={deleteItem}>
            close
          </i>
          <h3 className="card__type">{startCase(type)}</h3>
          <h4 className="card__duration">{`${duration} hr${
            duration > 1 ? 's' : ''
          }`}</h4>
          <h3 className="card__severity">{startCase(severity)}</h3>
        </div>
      )}
    </Draggable>
  )
}

Card.propTypes = {
  id: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  type: PropTypes.oneOf(['feature', 'bugFix', 'update', 'research', 'content'])
    .isRequired,
  duration: PropTypes.number.isRequired,
  severity: PropTypes.oneOf(['hight', 'medium', 'low']).isRequired,
  TaskListIndex: PropTypes.number.isRequired
}

export default React.memo(Card)
