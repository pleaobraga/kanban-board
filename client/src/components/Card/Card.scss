@import '../../styles/_color.scss';

.card {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: space-around;

  clip-path: polygon(50% 0%, 100% 0, 100% 88%, 81% 100%, 0 100%, 0 0);

  position: relative;

  width: 21rem;
  height: 15rem;

  padding: 0 2.2rem;

  &:hover .close-icon {
    display: block;
  }

  .close-icon {
    position: absolute;
    top: 5px;
    right: 6px;
    display: none;

    font-size: 3rem;

    width: 3rem;
    height: 3rem;

    cursor: pointer;
  }

  &:before {
    content: '';

    position: absolute;
    bottom: 0;
    right: 0;
    display: block;

    width: 4rem;
    height: 2rem;

    background: black;
    opacity: 0.2;
  }

  &--feature {
    background: $blue-light;
  }

  &--update {
    background-color: $orange;
  }

  &--bug-fix {
    background-color: $salmon;
  }

  &--research {
    background-color: $purple;
  }

  &--content {
    background-color: $green;
  }

  &--dragging {
    background-color: lightgreen;
  }

  &__type {
    font-family: 'Handlee', cursive;
    font-size: 4rem;
    color: $dark-blue;
    text-align: left;
  }

  &__duration {
    font-family: 'Handlee', cursive;
    font-size: 3rem;
    color: $white;
    text-align: left;
  }

  &__severity {
    font-family: 'Handlee', cursive;
    font-size: 3rem;
    color: $white;
    text-align: cenleftter;
  }
}
