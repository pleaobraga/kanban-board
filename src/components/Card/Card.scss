@import '../../styles/_colors.scss';

.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;

  clip-path: polygon(50% 0%, 100% 0, 100% 88%, 81% 100%, 0 100%, 0 0);

  position: relative;

  width: 20rem;
  height: 15rem;

  &:before {
    content: '';

    position: absolute;
    bottom: 0;
    right: 0;
    display: block;

    width: 4rem;
    height: 2rem;

    background: black;
    opacity: 0.2;
  }

  &__feature {
    background: $blue;
  }

  &__update {
    background-color: $orange;
  }

  &__bug-fix {
    background-color: $salmon;
  }

  &__research {
    background-color: $salmon;
  }

  &__content {
    background-color: $green;
  }

  &--type {
    font-family: 'Handlee', cursive;
    font-size: 4rem;
    color: $dark-blue;
    text-align: left;
  }

  &--duration {
    font-family: 'Handlee', cursive;
    font-size: 3rem;
    color: $white;
    text-align: left;
  }

  &--severity {
    font-family: 'Handlee', cursive;
    font-size: 3rem;
    color: $white;
    text-align: cenleftter;
  }
}
