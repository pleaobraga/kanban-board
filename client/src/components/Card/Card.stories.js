import React from 'react'
import { storiesOf } from '@storybook/react'
import { number, select } from '@storybook/addon-knobs'
import { DragDropContext, Droppable } from 'react-beautiful-dnd'

import ProviderWrapper from '../../../.storybook/provider'
import store from '../../../.storybook/configureStore'
import { board } from '../../reducers/__mocks__/reduxMock'
import { Card } from './Card'

const initialState = {
  board,
  loadingContent: false,
  errorContent: false
}

const newStore = store(initialState)

const withProvider = story => (
  <ProviderWrapper store={newStore}>{story()}</ProviderWrapper>
)

const getContent = () => {}

export default storiesOf('Components | Card', module)
  .addDecorator(withProvider)
  .add(
    'default',
    () => (
      <DragDropContext>
        <Droppable droppableId="droppable-1" type="PERSON">
          {provided => (
            <div ref={provided.innerRef} {...provided.droppableProps}>
              <Card
                getContent={getContent}
                id={'0'}
                index={0}
                type={select(
                  'Type',
                  ['feature', 'bug fix', 'update', 'research', 'content'],
                  'feature'
                )}
                duration={number('Duration', 1)}
                severity={select(
                  'Severity',
                  ['hight', 'medium', 'low'],
                  'medium'
                )}
                TaskListIndex={1}
              />
            </div>
          )}
        </Droppable>
      </DragDropContext>
    ),
    {
      info: {
        inline: true,
        header: false,
        source: false,
        propTables: [Card]
      }
    }
  )
