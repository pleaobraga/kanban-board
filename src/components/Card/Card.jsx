import './Card.scss'

import React from 'react'
import PropTypes from 'prop-types'
import { startCase, kebabCase } from 'lodash'
import { Draggable } from 'react-beautiful-dnd'

export const Card = ({ id, index, type, duration, severity }) => {
  const deleteItem = () => {
    //TODO Implemen delete function
  }

  return (
    <Draggable draggableId={id} index={index}>
      {provided => (
        <div
          className={`card card__${kebabCase(type)}`}
          ref={provided.innerRef}
          {...provided.draggableProps}
          {...provided.dragHandleProps}
        >
          <i className="material-icons close-icon" onClick={deleteItem}>
            close
          </i>
          <h3 className="card--type">{startCase(type)}</h3>
          <h4 className="card--duration">{`${duration} hr${
            duration > 1 ? 's' : ''
          }`}</h4>
          <h3 className="card--severity">{startCase(severity)}</h3>
        </div>
      )}
    </Draggable>
  )
}

Card.propTypes = {
  id: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  type: PropTypes.oneOf(['feature', 'bugFix', 'update', 'research', 'content'])
    .isRequired,
  duration: PropTypes.number.isRequired,
  severity: PropTypes.oneOf(['hight', 'medium', 'low']).isRequired
}

export default React.memo(Card)
