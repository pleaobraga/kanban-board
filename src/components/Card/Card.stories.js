import React from 'react'
import { storiesOf } from '@storybook/react'
import { number, select } from '@storybook/addon-knobs'
import { Card } from './Card'

export default storiesOf('Components | Card', module).add(
  'default',
  () => (
    <Card
      type={select(
        'Type',
        ['feature', 'bug fix', 'update', 'research', 'content'],
        'feature'
      )}
      duration={number('Duration', 1)}
      severity={select('Severity', ['hight', 'medium', 'low'], 'medium')}
    />
  ),
  { info: { inline: true, header: false } }
)
