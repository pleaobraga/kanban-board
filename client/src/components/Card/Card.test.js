import React from 'react'
import { mount } from 'enzyme'
import Card from './Card'
import { Provider } from 'react-redux'
import { DragDropContext, Droppable } from 'react-beautiful-dnd'
import {
  mockStore,
  initialStateRootReducer,
  card
} from '../../reducers/__mocks__/reduxMock'

const setup = ({ props = {}, state = {} }) => {
  const newProps = {
    ...card,
    ...props
  }

  const newState = {
    board: {
      ...initialStateRootReducer.content,
      ...state
    }
  }

  const store = mockStore(newState)

  const contentPage = mount(
    <Provider store={store}>
      <DragDropContext>
        <Droppable droppableId="droppable-1" type="PERSON">
          {provided => (
            <div ref={provided.innerRef} {...provided.droppableProps}>
              <Card {...newProps} />
            </div>
          )}
        </Droppable>
      </DragDropContext>
    </Provider>
  )

  return contentPage
}

describe('Card', () => {
  it('should render properly', () => {
    const component = setup({})
    expect(component).toMatchSnapshot()
  })
})
