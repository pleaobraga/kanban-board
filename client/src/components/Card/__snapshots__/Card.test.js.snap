// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <DragDropContext>
    <ErrorBoundary>
      <App
        contextId="0"
        liftInstruction="Draggable item. Ensure your screen reader is not in browse mode and then press space bar to lift."
        setCallbacks={[Function]}
      >
        <Provider
          context={
            Object {
              "$$typeof": Symbol(react.context),
              "Consumer": Object {
                "$$typeof": Symbol(react.context),
                "_calculateChangedBits": null,
                "_context": [Circular],
              },
              "Provider": Object {
                "$$typeof": Symbol(react.provider),
                "_context": [Circular],
              },
              "_calculateChangedBits": null,
              "_currentRenderer": Object {},
              "_currentRenderer2": null,
              "_currentValue": null,
              "_currentValue2": null,
              "_threadCount": 0,
            }
          }
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
        >
          <ConnectFunction
            direction="vertical"
            droppableId="droppable-1"
            getContainerForClone={[Function]}
            ignoreContainerClipping={false}
            isCombineEnabled={false}
            isDropDisabled={false}
            mode="standard"
            renderClone={null}
            type="PERSON"
          >
            <Droppable
              direction="vertical"
              droppableId="droppable-1"
              getContainerForClone={[Function]}
              ignoreContainerClipping={false}
              isCombineEnabled={false}
              isDropDisabled={false}
              mode="standard"
              placeholder={null}
              renderClone={null}
              shouldAnimatePlaceholder={false}
              snapshot={
                Object {
                  "draggingFromThisWith": null,
                  "draggingOverWith": null,
                  "isDraggingOver": false,
                  "isUsingPlaceholder": false,
                }
              }
              type="PERSON"
              updateViewportMaxScroll={[Function]}
              useClone={null}
            >
              <div
                data-rbd-droppable-context-id="0"
                data-rbd-droppable-id="droppable-1"
              >
                <Memo(Card)
                  TaskListId="ae0e3de4-fa5c-4793-bde8-6c0b77a146fc"
                  TaskListIndex={0}
                  duration={2}
                  id="448957e7-99e0-41d2-b36d-7b647063293d"
                  index={0}
                  severity="hight"
                  type="feature"
                >
                  <PublicDraggable
                    draggableId="448957e7-99e0-41d2-b36d-7b647063293d"
                    index={0}
                  >
                    <PrivateDraggable
                      canDragInteractiveElements={false}
                      draggableId="448957e7-99e0-41d2-b36d-7b647063293d"
                      index={0}
                      isClone={false}
                      isEnabled={true}
                      shouldRespectForcePress={false}
                    >
                      <Connect(Draggable)
                        canDragInteractiveElements={false}
                        draggableId="448957e7-99e0-41d2-b36d-7b647063293d"
                        index={0}
                        isClone={false}
                        isEnabled={true}
                        shouldRespectForcePress={false}
                      >
                        <Draggable
                          canDragInteractiveElements={false}
                          draggableId="448957e7-99e0-41d2-b36d-7b647063293d"
                          dropAnimationFinished={[Function]}
                          index={0}
                          isClone={false}
                          isEnabled={true}
                          mapped={
                            Object {
                              "combineTargetFor": null,
                              "offset": Object {
                                "x": 0,
                                "y": 0,
                              },
                              "shouldAnimateDisplacement": true,
                              "snapshot": Object {
                                "combineTargetFor": null,
                                "combineWith": null,
                                "draggingOver": null,
                                "dropAnimation": null,
                                "isClone": false,
                                "isDragging": false,
                                "isDropAnimating": false,
                                "mode": null,
                              },
                              "type": "SECONDARY",
                            }
                          }
                          shouldRespectForcePress={false}
                        >
                          <div
                            aria-labelledby="rbd-lift-instruction-0"
                            className="card card--feature false"
                            data-rbd-drag-handle-context-id="0"
                            data-rbd-drag-handle-draggable-id="448957e7-99e0-41d2-b36d-7b647063293d"
                            data-rbd-draggable-context-id="0"
                            data-rbd-draggable-id="448957e7-99e0-41d2-b36d-7b647063293d"
                            draggable={false}
                            onDragStart={[Function]}
                            onTransitionEnd={null}
                            style={
                              Object {
                                "transform": null,
                                "transition": null,
                              }
                            }
                            tabIndex={0}
                          >
                            <i
                              className="material-icons close-icon"
                              onClick={[Function]}
                            >
                              close
                            </i>
                            <h3
                              className="card__type"
                            >
                              Feature
                            </h3>
                            <h4
                              className="card__duration"
                            >
                              2 hrs
                            </h4>
                            <h3
                              className="card__severity"
                            >
                              Hight
                            </h3>
                          </div>
                        </Draggable>
                      </Connect(Draggable)>
                    </PrivateDraggable>
                  </PublicDraggable>
                </Memo(Card)>
              </div>
            </Droppable>
          </ConnectFunction>
        </Provider>
      </App>
    </ErrorBoundary>
  </DragDropContext>
</Provider>
`;
